sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsConnectionFactoryConfiguration.java

```diff
@@ -33,8 +33,7 @@ private static ServiceBusJmsConnectionFactory createJmsConnectionFactory(AzureSe
     }
 
     @Configuration(proxyBeanMethods = false)
-    @ConditionalOnProperty(prefix = "spring.jms.servicebus.pool", name = "enabled", havingValue = "false",
-        matchIfMissing = true)
+    @ConditionalOnProperty(prefix = "spring.jms.servicebus.pool", name = "enabled", havingValue = "false")
```

Comment:
Based on our discussion, we don't need to change the code in the configuration class, as it follows the order:
- If pool sets to true, use pool configuration, no matter what cache value is (true, false, null)
- If pool sets to false, or null, use simple configuration
  - If cache sets to true or null, use cache configuration
  - If cache sets to false, use ServiceBusJmsConnectionFactory

So, 
- if users set pool to true/false in their applications, respect them not matter what cache value is
- if users don't set pool in their applications
  - if cache is set to true, use cache
  - if cache is set to false, use service bus jms <- we can discuss whether to set pool in this case as well
  - if cache is not set, we set pool=true for them <- this is the change of default behavior